oauth-web-worker/README.md

```diff
@@ -0,0 +1,187 @@
+
+## How do I use this?
+
+The library exposes a singleton class called `OAuth`. Get an instance of the `OAuth` class by calling its `getInstance()` method.
+
+```javascript
+var oAuth = Wso2OAuth.OAuth.getInstance();
+```
+
+Then, initialize the instance by passing the config parameters.
+
+```javascript
+oAuth
+    .initialize({
+        clientHost: "https://localhost:9443/worker",
+        clientID: "70gph7I55ioGi5FqhLPz8JvxZCEa",
+        serverOrigin: "https://localhost:9443",
+        baseUrls: ["https://localhost:9443"],
+        origin: origin,
+        callbackURL: "https://localhost:9443/worker",
+        enablePKCE: true,
+        scope: ["SYSTEM", "openid"]
+    })
+    .then((response) => {
+        console.log(response);
+    })
+    .catch((error) => {
+        console.log(error);
+    });
+```
+
+The following is the configuration object that is passed into the initialize method.
+
+```typescript
+export interface ConfigInterface {
+    authorizationType?: string;
+    clientHost: string;
+    clientID: string;
+    clientSecret?: string;
+    consentDenied?: boolean;
+    enablePKCE?: boolean;
+    prompt?: string;
+    responseMode?: ResponseModeTypes;
+    scope?: string[];
+    serverOrigin: string;
+    baseUrls: string[];
+    callbackURL: string;
+}
+```
+
+Once initialized, you can sign in by calling the `signIn()` method.
+
+```javascript
+oAuth
+    .signIn()
+    .then((response) => {
+        console.log(response);
+    })
+    .catch((error) => {
+        console.error(error);
+    });
+```
+
+The `signIn()` method listens to the authorization code by itself. But if the component rendered by the callback URL is different to the sign-in component, use the `listenForAuthCode()` method to capture the authorization code and continue with the authentication flow.
+
+```javascript
+oAuth
+    .listenForAuthCode()
+    .then((response) => console.log(response))
+    .catch((error) => console.log(error));
+```
+
+To sign out,
+
+```javascript
+oAuth
+    .signOut()
+    .then((response) => {
+        console.log(response);
+    })
+    .catch((error) => console.log(error));
+```
+
+To make an API call,
+
+```javascript
+const requestConfig = {
+    headers: {
+        "Access-Control-Allow-Origin": origin,
+        "Content-Type": "application/json"
+    },
+    method: "GET",
+    url: "https://localhost:9443/scim2/Me"
+};
+
+oAuth
+    .httpRequest(requestConfig)
+    .then((response) => {
+        console.log(response);
+    })
+    .catch((error) => {
+        console.log(error);
+    });
+```
+
+To make multiple api calls at once (this is similar to `axios.all` or `Promise.all`),
+
+```javascript
+const headers = {
+    Accept: "application/json",
+    "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost
```

Comment:
This is only applicable for redux based projects and how the app is structured. Let's have a standard string here